project/src/main/kotlin/org/ionproject/core/course/CourseSpringController.kt

```diff
@@ -0,0 +1,60 @@
+package org.ionproject.core.course
+
+import org.ionproject.core.common.*
+import org.ionproject.core.course.representations.courseToDetailRepr
+import org.ionproject.core.course.representations.courseToListRepr
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.*
+import java.util.*
+
+/*
+ * Course controller, accepts only application/json and vnd.siren+json,
+ * should it accept any type and return an error in case there is no
+ * representation?
+ */
+@RestController
+class CourseSpringController(private val courseServices: CourseServices){
+
+
+    @GetMapping(Uri.courses)
+    fun getCourses(@RequestParam page : Optional<Int>, @RequestParam limit : Optional<Int>) : ResponseEntity<Siren> {
+        val page = page.orElseGet { 0 }
+        val limit = limit.orElseGet { 5 }           //Default limit is 5 records
```

Comment:
You can replace these with `@RequestParam(defaultValue = "0") page: Int`

Reply:
If a value is not used, doesn't that mean that the default value string "0" will need to pass an additional cast to Int, while Optional<Int> only accepts Int's?
If not, whats the advantage of using "@RequestParam(defaultValue = "0")" instead of Optional?

Reply:
Because I'm lazy and don't want to have to write `val page = page.orElseGet { 0 }; val limit = limit.orElseGet { 5 }` every time 😂.

"while Optional only accepts Int's?" `page: Int` --> has an `Int` type annotation and `defaultValue = "0"` has a hardcoded number.